### PR TITLE
reporting API

### PR DESCRIPTION
**Pull request by wabinyai**
**_WHAT DOES THIS PR DO?_**
The API intend to easy reporting for each grid
Example how to run 
http://127.0.0.1:5000/api/v2/analytics/grid/report
{
    "grid_id": "64b7ba65d7249f0029fecdbd",
    "start_time": "2023-01-01T00:00",
    "end_time": "2023-12-30T00:00"
}
 
- [ ] Item 1
- [ ] Item 2

**_WHAT ISSUES ARE RELATED TO THIS PR?_**

- Jira cards
    - []
- GitHub issues
    - Closes #<enter GitHub issue number here>

**_HOW DO I TEST OUT THIS PR?_**

- [x] Analytics: [Link to Analytics README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/analytics#readme)
- [ ] Auth Service: [Link to Auth Service README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/auth-service#readme)
- [ ] Calibrate: [Link to Calibrate README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/calibrate#readme)
- [ ] Data Proxy: [Link to Data Proxy README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/data-proxy#readme)
- [ ] Device Monitoring: [Link to Device Monitoring README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/device-monitoring#readme)
- [ ] Device Registry: [Link to Device Registry README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/device-registry#readme)
- [ ] Device Status: [Link to Device Status README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/device-status#readme)
- [ ] Device Uptime: [Link to Device Uptime README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/device-uptime#readme)
- [ ] Exceedances: [Link to Exceedances README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/exceedances#readme)
- [ ] Firebase: [Link to Firebase README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/firebase)
- [ ] GP Model: [Link to GP Model README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/gp-model#readme)
- [ ] Incentives: [Link to Incentives README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/incentives#readme)
- [ ] Kafka Connectors: [Link to Kafka Connectors](https://github.com/airqo-platform/AirQo-api/tree/staging/src/kafka-connectors)
- [ ] Locate: [Link to Locate README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/locate#readme)
- [ ] Metadata: [Link to Metadata README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/meta-data#readme)
- [ ] Predict: [Link to Predict README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/predict#readme)
- [ ] Spatial: [Link to Spatial](https://github.com/airqo-platform/AirQo-api/tree/staging/src/spatial)
- [ ] View: [Link to View README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/view#readme)
- [ ] Workflows: [Link to Workflows README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/workflows#readme)

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

- [ ] Endpoint 1
- [ ] Endpoint 2

**_ARE THERE ANY RELATED PRs?_**

- [ ] Related PR 1
- [ ] Related PR 2

### CODE REVIEW

**Review by Baalmart — COMMENTED** [review]
Hi @wabinyai , thanks for this.
Please note that this PR has some merge conflicts that need to be resolved.
Please resolve them accordingly.

**Review by sserurich — CHANGES_REQUESTED** [review]
- ensure the url on line 24 is read from the config file,
-update the naming for the routes as discussed

**Comment by wabinyai** [comment]
> * ensure the url on line 24 is read from the config file,
>   -update the naming for the routes as discussed

This was resolved

**Comment by wabinyai** [comment]
> While the time is local, the GMT should also reflect the local GMT. This is for Lagos and we are on GMT+1
> 
> image

The issue has been resolved instead of GMT it shows +01 for Nigeria +0300 for Uganda
image
image

**Review by BenjaminSsempala — APPROVED** [review]

**Review by uman95 — APPROVED** [review]
Works fine

**Review by Baalmart — APPROVED** [review]

File: src/analytics/config.py [hunk 1]
```diff
@@ -21,6 +21,8 @@ class Config:
     TESTING = False
     CSRF_ENABLED = True
     SECRET_KEY = env_var("SECRET_KEY")
+    GRID_URL = "https://platform.airqo.net/api/v2/devices/measurements/grids/"
```
Comment (Baalmart) [inline src/analytics/config.py:24]: Hi @wabinyai , please have this URL read from a config file.

**Review by Baalmart — CHANGES_REQUESTED** [review]
Hi @wabinyai , just have a small review comment....similar to what @sserurich already requested. Appears it is not yet resolved.

**Review by Baalmart — APPROVED** [review]

**Pull request merged**